#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_ciddr_ftbfs.dpatch by Stefan Potyra <sistpoty@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: do not overwrite the const argument, but rather make use of the
## DP: buffer in ciddr::set

@DPATCH@
Index: libcommoncpp2-1.8.1/src/cidr.cpp
===================================================================
--- libcommoncpp2-1.8.1.orig/src/cidr.cpp	2010-11-01 10:15:28.000000000 +1100
+++ libcommoncpp2-1.8.1/src/cidr.cpp	2012-05-19 10:52:26.000000000 +1000
@@ -199,7 +199,7 @@
     bitset((bit_t *)&netmask, getMask(cp));
     setString(cbuf, sizeof(cbuf), cp);
 
-    ep = (char *)strchr(cp, '/');
+    ep = (char *)strchr(cbuf, '/');
 
     if(ep)
         *ep = 0;
@@ -328,7 +328,7 @@
     memset(&netmask, 0, sizeof(netmask));
     bitset((bit_t *)&netmask, getMask(cp));
     setString(cbuf, sizeof(cbuf), cp);
-    ep = (char *)strchr(cp, '/');
+    ep = (char *)strchr(cbuf, '/');
     if(ep)
         *ep = 0;
 
